pkg/sql/delegate/show_roles.go

```diff
@@ -10,10 +10,14 @@
 
 package delegate
 
-import "github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqltelemetry"
+)
 
 // delegateShowRoles implements SHOW ROLES which returns all the roles.
 // Privileges: SELECT on system.users.
 func (d *delegator) delegateShowRoles(n *tree.ShowRoles) (tree.Statement, error) {
+	sqltelemetry.IncIAMShow(sqltelemetry.ROLE)
```

Comment:
I think you should to the `show` style telemetry for this command, because it _is_ a show command. `sqltelemetry/show.go`